src/main/java/org/pentaho/di/trans/dataservice/serialization/ServiceTrans.java

```diff
@@ -221,12 +244,38 @@ public TransMeta load( Repository repository ) throws KettleException {
         if ( rd == null ) {
           rd = root; // root
         }
-        return repository.loadTransformation( repository.getTransformationID( name, rd ), null );
+
+        ObjectId transId = repository.getTransformationID( name, rd );
```

Comment:
When disconnected from the repo, I discovered I had one data service defined in my file-based metastore that had a REPO_PATH reference listed (instead of FILE).  Not sure how it got in there with REPO_PATH, afaik that shouldn't happen.  But since it did, got a NPE on the line above when trying to retrieve it from a null repository.  Might be good to have a defensive null check with an error or warn log.